workflows/sampletracking.nf

```diff
@@ -40,12 +43,55 @@ workflow SAMPLETRACKING {
     def ch_multiqc_files = Channel.empty()
     def ch_pool_multiqc_files = Channel.empty()
 
+    def (ch_bams_with_index, ch_bams_no_index) = ch_samplesheet
+    .branch {
+        with_index: { meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index -> sample_bam_index != null }
+        no_index:  { meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index -> sample_bam_index == null }
+    }
+
+    def ch_bams_to_index = ch_bams_no_index
+        .map { meta, sample_bam, _sample_bam_index, _snp_fastq, _snp_bam, _snp_bam_index -> [meta, sample_bam] }
+
+    SAMTOOLS_INDEX(ch_bams_to_index)
+    ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())
+
+    def ch_bams_no_index_fixed = ch_bams_no_index
+        .join(SAMTOOLS_INDEX.out.bai, by: 0)
+        .map { meta, sample_bam, _sample_bam_index, snp_fastq, snp_bam, snp_bam_index, sample_bam_index_new ->
+            [meta, sample_bam, sample_bam_index_new, snp_fastq, snp_bam, snp_bam_index]
+        }
+
+    def ch_samplesheet_fixed = ch_bams_with_index.mix(ch_bams_no_index_fixed)
+
+    def (ch_snp_bams_with_index, ch_snp_bams_no_index) = ch_samplesheet_fixed
```

Comment:
Could you try implementing it so that the indexing of both the files happens in parallel instead of serial? 

So now your code runs like this:
```
INPUT -> INDEX_BAM -> INDEX_SNP_BAM
```

We would expect the last two processes to run in parallel like this:
```
INPUT -> INDEX_BAM
INPUT -> INDEX_SNP_BAM
```

Feel free to ask me if you need any nudges in the right direction for this.